rbd/encryption.go

```diff
@@ -0,0 +1,116 @@
+// +build !octopus,!nautilus
+
+package rbd
+
+// #cgo LDFLAGS: -lrbd
+// /* force XSI-complaint strerror_r() */
+// #define _POSIX_C_SOURCE 200112L
+// #undef _GNU_SOURCE
+// #include <errno.h>
+// #include <stdlib.h>
+// #include <rados/librados.h>
+// #include <rbd/librbd.h>
+import "C"
+
+import (
+	"unsafe"
+)
+
+// cData contains the data needed by the encryption functions
+type cData struct {
```

Comment:
On second thought, since this is global within the rbd package, a more specific name would have been better. But we can always change that later when we experience a conflict. No need to push another version only for that.

Reply:
I have another PR ready for the rbd_encryption_load() api which I plan to submit as soon as we're done with this.  I can update the struct name as part of it, if that's your preference.